src/useIsShake.ts

```diff
@@ -51,14 +55,41 @@ export function useIsShake() {
     setSubscription(null);
   };
 
+  const checkAccelerometerAvailablity = async () => {
+    const availability = await isAvailableAsync();
+    setIsAccelerometerAvailable(() => {
+      setIsAccelerometerAvailPending(false);
+      return availability;
+    });
+  };
+
+  const getAccelerometerPermission = async () => {
+    const permissionResponse = await requestPermissionsAsync();
+    if (permissionResponse.granted) {
+      setIsAccelerometerAvailable(() => {
+        setIsAccelerometerAvailPending(false);
+        return permissionResponse.granted;
+      });
+      // following else if block for when permissionResponse.canAskAgain === false in order to direct end user to Settings app in order to enable permission to access Accelerometer
+    } else if (!permissionResponse.canAskAgain) openSettings();
+  };
+
   useEffect(() => {
-    // invocation of _subscribe when isToggleReady === true
-    if (isShakeReady) {
+    if (isAccelerometerAvailable === 'unckecked' && !isAccelerometerAvailPending) {
+      setIsAccelerometerAvailPending(true);
+      checkAccelerometerAvailablity();
+    };
+    if (!isAccelerometerAvailable && !isAccelerometerAvailPending) {
+      setIsAccelerometerAvailPending(true);
+      getAccelerometerPermission();
+    };
```

Comment:
Logic flow to check for accelerometer availability, get permission, and/or direct to enable permission.